chain/events/filter/index.go

```diff
@@ -572,9 +617,27 @@ func (ei *EventIndex) prefillFilter(ctx context.Context, f *eventFilter, exclude
 		return xerrors.Errorf("exec prefill query: %w", err)
 	}
 
-	var ces []*CollectedEvent
+	// create struct for holding EventEntry with ordering
+	type eventEntryWithOrdering struct {
+		types.EventEntry
+		ordering uint64
```

Comment:
I don't think we need to select this field, we just need to order by it when querying.

Reply:
@Stebalien For some reason, if I add this to the ordering, we get way more records than we want and the test starts failing. I spent time debugging it before giving up. 

I had changed the current ordering to  	s += "ORDER BY event.height DESC, event_entry.ordering ASC".